File: VB/WindowsApplication1/XtraForm1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraGauges.Win.Gauges.Linear
Imports DevExpress.XtraGauges.Core.Drawing
Imports DevExpress.XtraGauges.Core.Model

Namespace WindowsApplication1
	Partial Public Class XtraForm1
		Inherits DevExpress.XtraEditors.XtraForm
		Public Sub New()
			InitializeComponent()
		End Sub

		Private time As Integer
		Private Const maxtime As Integer = 50
		Private Const highlightedIndicatorsCount As Integer = 4
		Private opacityValues() As Single = {0.5F, 0.7F, 0.9F, 1F }

		Public Sub ShowProgress()
			ShowDialog()
		End Sub

		Private Shared Sub HighlightIndicator(ByVal indicator As LinearScaleStateIndicatorComponent, ByVal opacity As Single)
			Dim shader As New OpacityShader()
			shader.Opacity = opacity
			indicator.Shader = shader
			indicator.States(0).ShapeType = StateIndicatorShapeType.ProgressItem9
		End Sub


		Public Sub SetProgress(ByVal progress As Integer)
			progress = progress Mod linearGauge1.Indicators.Count
			Dim highlightedIndicators(highlightedIndicatorsCount - 1) As LinearScaleStateIndicatorComponent
			For i As Integer = 0 To highlightedIndicatorsCount - 1
				highlightedIndicators(i) = linearGauge1.Indicators((progress + i) Mod linearGauge1.Indicators.Count)
			Next i

			For Each indicator As LinearScaleStateIndicatorComponent In linearGauge1.Indicators
				indicator.Shader = BaseColorShader.Empty
				indicator.States(0).ShapeType = StateIndicatorShapeType.ProgressItem5
			Next indicator
			For i As Integer = 0 To highlightedIndicatorsCount - 1
				HighlightIndicator(highlightedIndicators(i), opacityValues(i))
			Next i
		End Sub

		Private Sub timer1_Elapsed(ByVal sender As Object, ByVal e As System.Timers.ElapsedEventArgs) Handles timer1.Elapsed
			SetProgress(time)
			time += 1
			If time > maxtime Then
				timer1.Stop()
				Dispose()
			End If
		End Sub
	End Class
End Namespace

File: CS/WindowsApplication1/XtraForm1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraGauges.Win.Gauges.Linear;
using DevExpress.XtraGauges.Core.Drawing;
using DevExpress.XtraGauges.Core.Model;

namespace WindowsApplication1
{
    public partial class XtraForm1 : DevExpress.XtraEditors.XtraForm
    {
        public XtraForm1()
        {
            InitializeComponent();
        }

        int time;
        const int maxtime = 50;
        const int highlightedIndicatorsCount = 4;
        float[] opacityValues = new float[] {0.5F, 0.7F, 0.9F, 1F };

        public void ShowProgress()
        {
            ShowDialog();
        }

        static void HighlightIndicator(LinearScaleStateIndicatorComponent indicator, float opacity)
        {
            OpacityShader shader = new OpacityShader();
            shader.Opacity = opacity;
            indicator.Shader = shader;
            indicator.States[0].ShapeType = StateIndicatorShapeType.ProgressItem9;
        }


        public void SetProgress(int progress)
        {
            progress = progress % linearGauge1.Indicators.Count;
            LinearScaleStateIndicatorComponent[] highlightedIndicators = new LinearScaleStateIndicatorComponent[highlightedIndicatorsCount];
            for (int i = 0; i < highlightedIndicatorsCount; i++)
            {
                highlightedIndicators[i] = linearGauge1.Indicators[(progress + i) % linearGauge1.Indicators.Count];
            }

            foreach (LinearScaleStateIndicatorComponent indicator in linearGauge1.Indicators)
            {
                indicator.Shader = BaseColorShader.Empty;
                indicator.States[0].ShapeType = StateIndicatorShapeType.ProgressItem5;
            }
            for (int i = 0; i < highlightedIndicatorsCount; i++)
            {
                HighlightIndicator(highlightedIndicators[i], opacityValues[i]);
            }
        }

        private void timer1_Elapsed(object sender, System.Timers.ElapsedEventArgs e)
        {
            SetProgress(time);
            time++;
            if (time > maxtime)
            {
                timer1.Stop();
                Dispose();
            }
        }
    }
}